sdk/logical/response.go

```diff
@@ -202,9 +208,19 @@ type HTTPResponseWriter struct {
 // NewHTTPResponseWriter creates a new HTTPResponseWriter object that wraps the
 // provided io.Writer.
 func NewHTTPResponseWriter(w http.ResponseWriter) *HTTPResponseWriter {
-	return &HTTPResponseWriter{
-		ResponseWriter: w,
-		written:        new(uint32),
+	// FIXME: this is a band aid and not a good solution, just put it here
```

Comment:
I have a draft PR to address this issue in a better way here
https://github.com/hashicorp/vault/pull/13200